src/main/java/com/gooddata/executeafm/afm/SimpleMeasureDefinition.java

```diff
@@ -98,9 +125,44 @@ public String getUri() {
         return getItem().getUri();
     }
 
+    /**
+     * Returns the qualifier used by the {@link com.gooddata.md.Metric}.
+     *
+     * @return qualifier used by the metric, which is its {@link #item}.
+     *
+     * @throws IllegalStateException
+     *         The exception is thrown when {@link #item} has not been set.
+     * @deprecated Use {@link #getObjQualifiers()} instead.
+     */
     @Override
+    @Deprecated
     public ObjQualifier getObjQualifier() {
-        return getItem();
+        return getObjQualifiers().stream()
```

Comment:
changing behaviour and may be should be in default method?

Reply:
Fixed.

Reply:
I did not want to add default method as I would want default to throw exception in case when nothing is found. If method can return null, it should have `Optional` return type which would change the contract of the method.

Reply:
UPDATE: I changed it to fallback to new `getObjQualifiers` and returning the null. This way the deprecated method falls back to single implementation and in case of changes nothing would get forgotten with logic implemented in two places and I left the original behavior of the method unchanged.